text/0000-extract-extender.md

```diff
@@ -0,0 +1,264 @@
+- When `<build-image>` is provided (optional), the lifecycle:
+    - MUST record the digest reference to the provided `<build-image>` in `<analyzed>`
+    - MUST copy the OCI manifest and config file for `<build-image>` to `<kaniko-dir>/cache`
+- The lifecycle:
+    - **If** `<analyzed>` has `run-image.extend = true`, the lifecycle:
+        - MUST download from the registry and save in OCI layout format the `run-image` in `<analyzed>` to `<kaniko-dir>/cache`
+
+E.g., after the restorer has run, `<kaniko-dir>/cache/base/<run-image-identifier>/oci-layout` should exist. If we are not using kaniko then this may no longer be necessary!
+
+### extender
+
+* `<kaniko-dir>` is a required input
+* `<kaniko-cache-ttl>` is an optional input
+* `<kaniko-dir>/cache` is an output (the changes here are specific to kaniko)
+* The ability to extend the build image in-container and drop down to the CNB user to run the build phase could potentially be a kaniko-specific behavior that we might not be able to replicate with an alternative implementation.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  - It is insecure to rely on an unsupported dependency
+  - We're currently blocked on upgrading the docker dependency in the lifecycle because of kaniko
+- What use cases does it support?
+  - More flexibility for platforms to "bring your own" extender
+- What is the expected outcome?
+  - It should still be possible to use image extensions by using CNB-provided tooling
+
+# What it is
+[what-it-is]: #what-it-is
+
+- Target persona: platform operator, platform implementor, and/or project contributor.
+
+* Starting from lifecycle 0.21.0 (the next minor as of this writing), we should remove the extender from [cmd/lifecycle](https://github.com/buildpacks/lifecycle/tree/main/cmd/lifecycle)
+  * We could create [cmd/extender](https://github.com/buildpacks/lifecycle/tree/main/cmd) with its own go.mod
+  * This will allow us to upgrade the docker version used by the lifecycle
+* We should deliver the extender via a separate tarball on the GitHub releases page, and as a separate image at `buildpacksio/extender:<version>`. In both cases, the directory structure would look like the following:
+
+```
+Permission     UID:GID       Size  Filetree                                   
+drwxr-xr-x         0:0      32 MB  └── cnb                                    
+drwxr-xr-x         0:0      32 MB      ├── lifecycle                                     
+-rwxrwxrwx         0:0        0 B      │   ├── extender
+-rw-r--r--         0:0      10 kB      │   ├── extender.sbom.cdx.json         
+-rw-r--r--         0:0      16 kB      │   ├── extender.sbom.spdx.json        
+-rw-r--r--         0:0      15 kB      │   ├── extender.sbom.syft.json
+-rw-r--r--         0:0      560 B      └── lifecycle.toml
+```
+
+* When creating builders, pack will inspect the lifecycle version provided and pull the extender with the same version from the GitHub releases page.
```

Comment:
Actually, this is just as non-declarative as looking for "sibling" artifacts, and would prevent builder creation in offline environments. We should probably go the uri route...